lib/ui/highlight-manager.js

```diff
@@ -252,16 +257,35 @@ HighlightManager.prototype = {
         this.popup.ignoreNextMouseOver = true;
         resultsContainer.ensureIndexIsVisible(0);
         this.stealHighlight();
+        newIndex = -1;
       } else if (evt.forward && !recommendationVisible) {
-        this.popup.el.selectedIndex = 0;
+        newIndex = 0;
       } else {
-        this.popup.el.selectedIndex = selectedIndex - 1;
+        newIndex = selectedIndex - 1;
       }
+      this.popup.el.selectedIndex = newIndex;
 
     // If the highlighted result was in the middle of the list,
     // then move the highlight to one of its neighbors.
     } else {
-      this.popup.el.selectedIndex = evt.forward ? selectedIndex + 1 : selectedIndex - 1;
+      newIndex = evt.forward ? selectedIndex + 1 : selectedIndex - 1;
+      this.popup.el.selectedIndex = newIndex;
+    }
+
+    // Finally, figure out the new url, and tell the urlbar that the selection
+    // has changed.
+    let newUrl;
+    if (newIndex === -1) {
+      let urlEl = this.recommendation.el.querySelector('#universal-search-recommendation-url');
+      newUrl = urlEl && urlEl.textContent;
```

Comment:
Rather than use `textContent`, can we use a `data-` attribute for meaningful data?